### PR TITLE
SEARCH-1081 | continuous deployment

### PR DESCRIPTION
**Pull request by bradgarropy**
## Description
Enable continuous deployment to `npm` through GitHub Actions.

## Related Issue
[SEARCH-1081](https://jira.corp.magento.com/browse/SEARCH-1081)

## Motivation and Context
This is pretty mandatory for any project.

## How Has This Been Tested?
Very manually, please pardon the commit log.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+jobs:
+  release:
+    name: release
+    runs-on: ubuntu-18.04
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.inputs.branch }}
```
Comment (deloreyj) [inline .github/workflows/release.yml:17]: Does this end up being whatever branch you tagged?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "magento-data-layer-sdk",
-  "version": "0.0.10",
+  "name": "@adobe/magento-data-layer-sdk",
+  "version": "0.0.8",
```
Comment (deloreyj) [inline package.json:3]: Since this value is no longer managed by us, should we put some sort of placeholder here?

**Review by deloreyj — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (bradgarropy) [reply .github/workflows/release.yml]: Great catch! Changed it to `github.ref`.

File: package.json (hunk 2 above)
Comment (bradgarropy) [reply package.json:3]: Actually from what I understand `standard-version` uses `package.json` to determine the last released version.

> standard-version will then do the following:
> 
> - Retrieve the current version of your repository by looking at packageFiles[1], falling back to the last git tag.
> - bump the version in bumpFiles[1] based on your commits.
> - Generates a changelog based on your commits (uses conventional-changelog under the hood).
> - Creates a new commit including your bumpFiles[1] and updated CHANGELOG.
> - Creates a new tag with the new version number.

**Pull request merged**